### ISSUE TITLE
Section 508 Accessibility - Cannot assign H3 tag to Inline Element 

### ISSUE THREAD

**Issue by aglynn** [gwtproject/gwt#9256]
I believe this might be vaguely related to https://github.com/gwtproject/gwt/issues/2297, but definitely not directly related.

AppVet Code: /src/gov/nist/gwt/client/gui/AppVetPanel.java: line 1026, 300, 319, and 339.

Problem: 

Line 1026: GWT InlineLabel cannot be set to use an HTML Heading tag.

![inlinenoheading](https://cloud.githubusercontent.com/assets/1056178/11981082/692af5dc-a96d-11e5-9867-7a6f4b600cd1.PNG)

This relates to Section 508 of the Rehabilitation Act, which is to help persons with disabilities. Specifically, it is for 9.2 Web: Section Headings, 1194.31 (Functional performance) a (Blindness), b (Low vision) Baselines: B15 - A. Visually apparent headings are not programmatically identified (31(a), 31(b), B15)

That means that since we have a title like element that we should provide an H1/H2/H3 around it rather than simply a span or div so that we can indicate to visually impaired users that this is a title.

**Comment by aglynn** [comment]
I just wanted to follow up to see if this issue could be triaged? I know you guys have a lot on your plate so I appreciate any assistance. Thank you.

**Comment by tbroyer** [comment]
Maybe use an [`HTMLPanel`](http://www.gwtproject.org/javadoc/latest/com/google/gwt/user/client/ui/HTMLPanel.html#HTMLPanel%28java.lang.String, java.lang.String%29) instead?

**Comment by jonl-percsolutions-com** [comment]
This is not really a bug per say.  There would be two approaches to accomplish your desires, one extend InlineLabel or Label or create a custom label.  Which approach might be easier depends on what the ultimate HTML needs to look like.  Say you need the HTML to look like this: `<h3>Header value</h3>`, you should be able to use the following class:

```
public class HLabel extends Label {

  /**
   * Creates a HLabel widget that wraps an existing &lt;div&gt; or
   * &lt;span&gt; element.
   *
   * This element must already be attached to the document. If the element is
   * removed from the document, you must call
   * {@link RootPanel#detachNow(Widget)}.
   *
   * @param element the element to be wrapped
   */
  public static HLabel wrap(Element element) {
    // Assert that the element is attached.
    assert Document.get().getBody().isOrHasChild(element);

    HLabel label = new HLabel(element);

    // Mark it attached and remember it for cleanup.
    label.onAttach();
    RootPanel.detachOnWindowClose(label);

    return label;
  }

  /**
   * Creates an empty label.
   *
   * @param headingType, the type of heading from 1 to 6 inclusive
   *
   */
  public HLabel(int headingType) {
    super(Document.get().createHElement(headingType));
    setStyleName("gwt-HLabel");
  }

  /**
   * Creates a label with the specified text.
   *
   * @param headingType, the type of heading from 1 to 6 inclusive
   * @param text the new label's text
   */
  public HLabel(int headingType,String text) {
    this(headingType);
    setText(text);
  }

  /**
   * Creates a label with the specified text and direction.
   *
   * @param headingType, the type of heading from 1 to 6 inclusive
   * @param text the new label's text
   * @param dir the text's direction. Note: {@code Direction.DEFAULT} means
   *        direction should be inherited from the widget's parent element.
   */
  public HLabel(int headingType,String text, Direction dir) {
    this(headingType);
    setText(text, dir);
  }

  /**
   * Creates a label with the specified text and a default direction estimator.
   *
   * @param headingType, the type of heading from 1 to 6 inclusive
   * @param text the new label's text
   * @param directionEstimator A DirectionEstimator object used for automatic
   *          direction adjustment. For convenience,
   *          {@link Label#DEFAULT_DIRECTION_ESTIMATOR} can be used.
   */
  public HLabel(int headingType,String text, DirectionEstimator directionEstimator) {
    this(headingType);
    setDirectionEstimator(directionEstimator);
    setText(text);
  }

  /**
   * This constructor may be used by subclasses to explicitly use an existing
   * element. This element must be either a &lt;div&gt; &lt;span&gt; element.
   *
   * @param element the element to be used
   */
  protected HLabel(Element element) {
    // super(element) also asserts that element is either a &lt;div&gt; or
    // &lt;span&gt;.
    super(element);
  }
}
```

If you need something more complex like <h3><span>Header<span></h3> you could use HTMLPanel or a custom ui.xml or more overrides to the Label, if you want it to be a label class.
